File: code/jvm/im-repository/src/main/kotlin/im/pagination/PaginationRequest.kt
package im.pagination

/**
 * Represents a pagination request.
 *
 * Page indexing starts at 1 when using this class.
 *
 * @param page the page number
 * @param size the page size
 * @param sort the sort
 */
data class PaginationRequest(
    val page: Int,
    val size: Int,
    val sort: Sort? = null
)

File: code/js/src/Domain/pagination/PaginationRequest.ts
/**
 * Represents a pagination request.
 * @param page The page number.
 * @param size The number of items per page.
 * @param getCount Indicates whether to return the total count of items.
 */
export interface PaginationRequest {
    offset: number;
    limit: number;
    getCount: boolean;
}
